### ISSUE TITLE
3.4.0. - False local offline

### ISSUE THREAD

**Issue by Masterz69** [AlexxIT/SonoffLAN#1112]
Recently updated integration to 3.4.0 version.
All my devices are in LAN mode, ALL connected & powered (except 2). 

Just checked diagnostics dump - see attached.
[config_entry-sonoff-a0656de0df0bb579aa4ece0459ab70e2.json.txt](https://github.com/AlexxIT/SonoffLAN/files/10740157/config_entry-sonoff-a0656de0df0bb579aa4ece0459ab70e2.json.txt)

Log contains such a records on looks like periodical basis. 
See info:200 - it means device responded ?
```
2023-02-15 02:08:45 [D] 1000e61943 => Local4 | {'cmd': 'info'} <= info: 200
2023-02-15 02:08:45 [D] 10012fa101 => Local4 | {'cmd': 'info'} <= info: 200
2023-02-15 02:08:45 [D] 10010a5056 => Local4 | {'cmd': 'info'} <= info: 200
2023-02-15 02:08:45 [D] 100166cee5 => Local4 | {'cmd': 'info'} <= info: 200
2023-02-15 02:08:45 [D] 100166d6b6 => Local4 | {'cmd': 'info'} <= info: 200
2023-02-15 02:08:45 [D] 10010a526e => Local4 | {'cmd': 'info'} <= info: 200
2023-02-15 02:08:45 [D] 10010a4da0 => Local4 | {'cmd': 'info'} <= info: 200
2023-02-15 02:08:45 [D] 1000f21e74 => Local4 | {'cmd': 'info'} <= info: 200
2023-02-15 02:08:45 [D] 100169f2b5 => Local4 | {'cmd': 'info'} <= info: 200
2023-02-15 02:08:45 [D] 10012b9e27 => Local4 | {'cmd': 'info'} <= info: 200
2023-02-15 02:08:45 [D] 100169f2ad => Local4 | {'cmd': 'info'} <= info: 200
```
Despite of periodical status check - part of devices from this list have NO host attribute.
As I wrote - all my eWeLink account devices in LAN mode, all except 2 connected & powered.

In v3.3.1 only DualR3 devices missing host attribute, as not sending IP in ZC packet.

[sonoff-log_20230215_01.txt](https://github.com/AlexxIT/SonoffLAN/files/10740186/sonoff-log_20230215_01.txt)

**Comment by Masterz69** [comment]
As soon eWeLink app started - see all except 2 device online in LAN mode.

Found in integration log:
```
2023-02-15 04:59:56 [D] 100166d6b6 <= Local0 | {'online': False} | 
2023-02-15 04:59:56 [D] 10010a4da0 <= Local0 | {'online': False} | 
2023-02-15 04:59:56 [D] 100166cee5 <= Local0 | {'online': False} | 
2023-02-15 04:59:56 [D] 10010a526e <= Local0 | {'online': False} | 
2023-02-15 04:59:56 [D] 10012f59cd <= Local0 | {'online': False} | 
2023-02-15 04:59:56 [D] 10010a5056 <= Local0 | {'online': False} | 
```

But this is definitely wrong ^^, this devices locally online.

**Comment by Masterz69** [comment]
Attaching Bonjour browser screenshot - can see devices responding:
![image](https://user-images.githubusercontent.com/15225823/218962491-418e7830-6501-43d4-82ed-c33a6e56f93a.png)

**Comment by Masterz69** [comment]
Devices responds on 
```
2023-02-15 04:59:32 [D] 100166cee5 => Local4 | {'cmd': 'info'} <= info: 200
2023-02-15 04:59:32 [D] 10010a526e => Local4 | {'cmd': 'info'} <= info: 200
2023-02-15 04:59:32 [D] 100166d6b6 => Local4 | {'cmd': 'info'} <= info: 200
2023-02-15 04:59:32 [D] 10010a4da0 => Local4 | {'cmd': 'info'} <= info: 200
2023-02-15 04:59:32 [D] 100169f1e7 => Local4 | {'cmd': 'info'} <= info: 200
```
Then declared ar locally offline:
```
2023-02-15 04:59:56 [D] 100166d6b6 <= Local0 | {'online': False} | 
2023-02-15 04:59:56 [D] 10010a4da0 <= Local0 | {'online': False} | 
2023-02-15 04:59:56 [D] 100166cee5 <= Local0 | {'online': False} | 
2023-02-15 04:59:56 [D] 10010a526e <= Local0 | {'online': False} | 
2023-02-15 04:59:56 [D] 10012f59cd <= Local0 | {'online': False} | 
2023-02-15 04:59:56 [D] 10010a5056 <= Local0 | {'online': False} | 
```

On one device example. Then happens 
```
2023-02-15 05:00:06 [D] 100166cee5 => Local4 | {'cmd': 'info'} !! Timeout 10
2023-02-15 05:00:06 [D] 100166cee5 !! Local4 | Device offline
```

And see NO MORE local online checks for that device.

Looks like local online checks proceeded only for devices ALREADY being online.
And can not recover devices which declared as locally offline by any reason.

**Comment by Masterz69** [comment]
Done this change in my system and will check for a while.
In core\ewelink\_init_.py:
```
        if not device.get("host"):
-           return
+           # try default if device already declared as offline
+           device["host"] = f"ewelink_{did}.local.:8081"
```

IMHO it will allow to check online status for every and each device, ESPECIALLY for declared as ```offline``` - they really needs this check to recover status.

**Comment by Masterz69** [comment]
One more case tonight.
Device become offline. From that time local online not checked anymore.
Same time it responding on request:
```
--2023-02-16 09:19:54--  http://ewelink_100169f1e7.local:8081/zeroconf/info
Resolving ewelink_100169f1e7.local (ewelink_100169f1e7.local)... 192.168.1.163
Connecting to ewelink_100169f1e7.local (ewelink_100169f1e7.local)|192.168.1.163|:8081... connected.
```

**Comment by Masterz69** [comment]
Today log attached: [sonoff-log_20230217_01.txt](https://github.com/AlexxIT/SonoffLAN/files/10764367/sonoff-log_20230217_01.txt)

Couple of devices suddenly become inactive:
```
2023-02-17 09:05:51 [D] 10010a526e <= Local0 | {'online': False} | 
2023-02-17 09:05:51 [D] 10010a4da0 <= Local0 | {'online': False} | 
2023-02-17 09:05:51 [D] 10010a4f46 <= Local0 | {'online': False} | 
2023-02-17 09:05:51 [D] 100166d6b6 <= Local0 | {'online': False} | 
2023-02-17 09:05:51 [D] 100166cee5 <= Local0 | {'online': False} | 
2023-02-17 09:05:51 [D] 10012f59cd <= Local0 | {'online': False} | 
```

After that integration NOT polling / checking local status anymore.
But actually devices online, appears in eWeLink app as locally managed.

As well it responds, f.e.:
````
Fri Feb 17 09:29:34 EET 2023
➜  log wget http://ewelink_100166cee5.local.:8081/zeroconf/info
--2023-02-17 09:29:42--  http://ewelink_100166cee5.local.:8081/zeroconf/info
Resolving ewelink_100166cee5.local. (ewelink_100166cee5.local.)... 192.168.1.234
Connecting to ewelink_100166cee5.local. (ewelink_100166cee5.local.)|192.168.1.234|:8081... connected.
````

No changes in WiFi setting for a long time. 
Sudden removals started with 3.4.0. installed together with Ha Core 2023.2.4 (from 2023.1.7) on same HA OS version.

**Comment by Masterz69** [comment]
Every time I starting 3.4.0 version - in following hours get massive outage of devices, like:
```
2023-02-17 16:07:20 [D] 100171fd9d <= Local3 | {'switch': 'off', 'startup': 'off', 'pulseConfig': {'pulse': 'off', 'switch': 'off', 'pulseWidth': 500}, 'sledOnline': 'on', 'tempUnit': 0, 'timeZone': 3, 'sensorType': 'AM2301', 'currentTemperature': '3.2', 'currentHumidity': '92.6', 'tempCorrection': 0, 'humCorrection': 0, 'autoControlEnabled': 0, 'fwVersion': '1.1.0', 'rssi': -84} | 1153879
2023-02-17 16:07:23 [D] 10012f59cd <= Local0 | {'online': False} | 
2023-02-17 16:07:23 [D] 100166cee5 <= Local0 | {'online': False} | 
2023-02-17 16:07:23 [D] 100166d6b6 <= Local0 | {'online': False} | 
2023-02-17 16:07:23 [D] 10010a4da0 <= Local0 | {'online': False} | 
2023-02-17 16:07:23 [D] 10010a526e <= Local0 | {'online': False} | 
2023-02-17 16:07:23 [D] 10010a4f46 <= Local0 | {'online': False} | 
2023-02-17 16:07:23 [D] 10010a5056 <= Local0 | {'online': False} | 
2023-02-17 16:07:23 [D] 10012f8d1b <= Local0 | {'online': False} | 
2023-02-17 16:07:25 [D] 100171fd9d <= Local3 | {'switch': 'off', 'startup': 'off', 'pulseConfig': {'pulse': 'off', 'switch': 'off', 'pulseWidth': 500}, 'sledOnline': 'on', 'tempUnit': 0, 'timeZone': 3, 'sensorType': 'AM2301', 'currentTemperature': '3.2', 'currentHumidity': '92.7', 'tempCorrection': 0, 'humCorrection': 0, 'autoControlEnabled': 0, 'fwVersion': '1.1.0', 'rssi': -84} | 1153880
2023-02-17 16:07:30 [D] 100171fd9d <= Local3 | {'switch': 'off', 'startup': 'off', 'pulseConfig': {'pulse': 'off', 'switch': 'off', 'pulseWidth': 500}, 'sledOnline': 'on', 'tempUnit': 0, 'timeZone': 3, 'sensorType': 'AM2301', 'currentTemperature': '3.2', 'currentHumidity': '92.7', 'tempCorrection': 0, 'humCorrection': 0, 'autoControlEnabled': 0, 'fwVersion': '1.1.0', 'rssi': -84} | 1153881
2023-02-17 16:07:33 [D] 10012f8d1b => Local4 | {'cmd': 'info'} !! Timeout 10
2023-02-17 16:07:33 [D] 10012f8d1b !! Local4 | Device offline
2023-02-17 16:07:33 [D] 10010a526e => Local4 | {'cmd': 'info'} !! Timeout 10
2023-02-17 16:07:33 [D] 10010a526e !! Local4 | Device offline
2023-02-17 16:07:33 [D] 10010a5056 => Local4 | {'cmd': 'info'} !! Timeout 10
2023-02-17 16:07:33 [D] 10010a5056 !! Local4 | Device offline
2023-02-17 16:07:33 [D] 10010a4f46 => Local4 | {'cmd': 'info'} !! Timeout 10
2023-02-17 16:07:33 [D] 10010a4f46 !! Local4 | Device offline
2023-02-17 16:07:33 [D] 10010a4da0 => Local4 | {'cmd': 'info'} !! Timeout 10
2023-02-17 16:07:33 [D] 10010a4da0 !! Local4 | Device offline
2023-02-17 16:07:33 [D] 100166d6b6 => Local4 | {'cmd': 'info'} !! Timeout 10
2023-02-17 16:07:33 [D] 100166d6b6 !! Local4 | Device offline
2023-02-17 16:07:33 [D] 100166cee5 => Local4 | {'cmd': 'info'} !! Timeout 10
2023-02-17 16:07:33 [D] 100166cee5 !! Local4 | Device offline
2023-02-17 16:07:33 [D] 10012f59cd => Local4 | {'cmd': 'info'} !! Timeout 10
2023-02-17 16:07:33 [D] 10012f59cd !! Local4 | Device offline
```

After that events devices loose 'host' attribute and not available locally anymore.
All next tries of connect from integration fails as follows:
```
2023-02-17 16:43:47 [D] 100166d6b6 => Local4 | {'cmd': 'info'} !! Timeout 10
2023-02-17 16:43:47 [D] 100166d6b6 !! Local4 | Device offline
```
Till HA Core restart.

One of ideas was - devices starts to block requests from HA cause classify periodic checks as flood.
Tried to change check period from 15 to 60.
Later commented line in 'run_forever()' to stop periodic 'check_offline' calls.
Same staff - after hours part of devices become 'removed'.
And see no way to recover them till HA Core restart.
Looks 'flooding' hypothesis is not right.

As soon installed 3.3.1 - no more offline devices, all working perfect and operated locally for an hours in a row.

**Comment by Masterz69** [comment]
One more observation.
After HA OS upgrade to 5.0 IPv6 addresses appears in mDNS, f.e.:
```
--2023-02-18 20:37:08--  http://ewelink_100169f2ad.local.:8081/zeroconf/info
Resolving ewelink_100169f2ad.local. (ewelink_100169f2ad.local.)... fe80::96b5:55ff:fef7:f108
Connecting to ewelink_100169f2ad.local. (ewelink_100169f2ad.local.)|fe80::96b5:55ff:fef7:f108|:8081... failed: Invalid argument.
```
While IPv6 is disabled in HA setting. 
Sametime on PC with ONLY IPv4 enabled:
![image](https://user-images.githubusercontent.com/15225823/219882752-c2453089-b405-4031-afef-137f82612646.png)

**Comment by Masterz69** [comment]
After some days of perfect working 3.3.1. version again installed 3.4.0. 
Integration started at
```
2023-02-19 17:30:18 [D] SysInfo: {'installation_type': 'Home Assistant OS', 'version': '2023.2.5', 'dev': False, 'hassio': True, 'virtualenv': False, 'python_version': '3.10.7', 'docker': True, 'arch': 'aarch64', 'timezone': 'Europe/Riga', 'os_name': 'Linux', 'os_version': '5.15.84-v8', 'user': 'root', 'supervisor': '2023.01.1', 'host_os': 'Home Assistant OS 9.5', 'docker_version': '20.10.22', 'chassis': 'embedded', 'sonoff_version': '3.4.0 (f6860d4)'}
2023-02-19 17:30:24 [D] 19 devices loaded from Cloud
```
Then 3 devices "removed":
```
2023-02-19 18:06:18 [D] 1000f3e195 <= Local0 |  | {'online': False} | 
2023-02-19 18:06:18 [D] 10012f59cd <= Local0 |  | {'online': False} | 
2023-02-19 18:06:18 [D] 1000a08c7c <= Local0 |  | {'online': False} | 
```
From this point in time devices does not answer during ```check_offline``` calls:
```
2023-02-19 18:16:51 [D] 10012f59cd => Local4 | {'cmd': 'info'} !! Timeout 10
2023-02-19 18:16:51 [D] 10012f59cd !! Local4 | Device offline
```

When trying to send command to this device getting not timeout, but ```ClientOSError```:
```
2023-02-19 18:17:41 [D] 10012f59cd => Local4 | {'switch': 'off'}|  File "/config/custom_components/sonoff/core/ewelink/local.py", line 195, in send|    r = await self.session.post(|aiohttp.client_exceptions.ClientOSError: [Errno 104] Connection reset by peer|
2023-02-19 18:17:41 [D] 10012f59cd => Cloud4 | {'switch': 'off'} | 1676823461000
2023-02-19 18:17:41 [D] 10012f59cd => Cloud4 | 1676823461001
2023-02-19 18:17:42 [D] 10012f59cd <= Cloud3 | {'version': 8, 'sledOnline': 'on', 'switch': 'off', 'fwVersion': '3.7.0', 'rssi': -83, 'startup': 'off', 'init': 1, 'pulse': 'off', 'pulseWidth': 500, 'only_device': {'ota': 'success'}} | 1676823461001
```

See command executed through. IMHO if device can connect to cloud - it is connected to WiFi, have IP. 
But not working in LAN mode, looks like.

mDNS still resolvable, but connection going to be dropped (by HA side or device ???) f.e.:
```
➜  ~ wget http://ewelink_10012f59cd.local.:8081/zeroconf/info
--2023-02-19 18:24:13--  http://ewelink_10012f59cd.local.:8081/zeroconf/info
Resolving ewelink_10012f59cd.local. (ewelink_10012f59cd.local.)... 192.168.1.149
Connecting to ewelink_10012f59cd.local. (ewelink_10012f59cd.local.)|192.168.1.149|:8081... connected.
HTTP request sent, awaiting response... Read error (Connection reset by peer) in headers.
Retrying.
```

In parallel I started on different computer ZC ServiceBrowser.
Almost same as here provided: https://pypi.org/project/zeroconf/
It also detected services removal:
```
2023-02-19 18:06:21.868269  | Service eWeLink_1000f3e195._ewelink._tcp.local. REMOVED
2023-02-19 18:06:24.876016  | Service eWeLink_10012f59cd._ewelink._tcp.local. REMOVED
2023-02-19 18:06:27.874254  | Service eWeLink_1000a08c7c._ewelink._tcp.local. REMOVED
```

Still have no clue why it not happens when running 3.3.1.

Will run 3.3.1. again and ServiceBrowser in parallel for couple of days.
Results will be published.

**Comment by Masterz69** [comment]
One more interesting case.
Device was actually disconnected from network.
And still after 5 minutes ZC Browsers nor on HA host nor on my PC does not received REMOVED event....

```
2023-02-19 18:47:46 [D] 1000cf7638 <= Local3 | {'mainSwitch': 'on', 'deviceType': 'temperature', 'targets': [{'targetHigh': 5, 'reaction': {'switch': 'off'}}, {'targetLow': 2, 'reaction': {'switch': 'on'}}], 'switch': 'off', 'startup': 'off', 'pulse': 'off', 'pulseWidth': 500, 'sledOnline': 'on', 'sensorType': 'AM2301', 'currentHumidity': '76', 'currentTemperature': '3.9'} | 9

2023-02-19 18:49:10 [D] 1000cf7638 => Local4 | {'cmd': 'info'} !! Timeout 10
2023-02-19 18:49:10 [D] 1000cf7638 !! Local4 | Device offline

2023-02-19 18:49:17 [D] 1000cf7638 => Local4 | {'cmd': 'info'} !! Can't connect: Cannot connect to host 192.168.1.231:8081 ssl:default [Connect call failed ('192.168.1.231', 8081)]
2023-02-19 18:49:17 [D] 1000cf7638 !! Local4 | Device offline

2023-02-19 18:49:54 [D] 1000cf7638 <= Cloud3 | {'online': False} | None
```
Cloud offline message already received, but not REMOVED on ZC Browsers.

When ZC Browser on PC restarted - 1000cf7638 not appears anymore.

**Comment by Masterz69** [comment]
Monitored for some time, ignoring UPDATED events:
```
C:\Scripts\ZC>python.exe test.py
Press enter to exit...

2023-02-19 18:59:45.072291  | Service eWeLink_1000f21e74._ewelink._tcp.local. ADDED
2023-02-19 18:59:45.078482  | Service eWeLink_10012f8d1b._ewelink._tcp.local. ADDED
2023-02-19 18:59:45.080437  | Service eWeLink_100166cee5._ewelink._tcp.local. ADDED
2023-02-19 18:59:45.082392  | Service eWeLink_1000a08c7c._ewelink._tcp.local. ADDED
2023-02-19 18:59:45.085321  | Service eWeLink_10010a526e._ewelink._tcp.local. ADDED
2023-02-19 18:59:45.089222  | Service eWeLink_10010a4da0._ewelink._tcp.local. ADDED
2023-02-19 18:59:45.090198  | Service eWeLink_10010a4f46._ewelink._tcp.local. ADDED
2023-02-19 18:59:45.092154  | Service eWeLink_10012f59cd._ewelink._tcp.local. ADDED
2023-02-19 18:59:45.094102  | Service eWeLink_10010a5056._ewelink._tcp.local. ADDED
2023-02-19 18:59:45.115315  | Service eWeLink_1000e61943._ewelink._tcp.local. ADDED
2023-02-19 18:59:45.117267  | Service eWeLink_10012fa101._ewelink._tcp.local. ADDED
2023-02-19 18:59:45.121174  | Service eWeLink_100166d6b6._ewelink._tcp.local. ADDED
2023-02-19 18:59:46.142943  | Service eWeLink_100169f2b5._ewelink._tcp.local. ADDED
2023-02-19 18:59:46.148799  | Service eWeLink_10012b9e27._ewelink._tcp.local. ADDED
2023-02-19 18:59:46.221948  | Service eWeLink_100169f2ad._ewelink._tcp.local. ADDED
2023-02-19 18:59:46.226991  | Service eWeLink_100171fd9d._ewelink._tcp.local. ADDED
2023-02-19 18:59:48.313009  | Service eWeLink_100169f1e7._ewelink._tcp.local. ADDED
2023-02-19 19:18:38.112724  | Service eWeLink_10012f8d1b._ewelink._tcp.local. REMOVED
2023-02-19 19:18:41.105374  | Service eWeLink_1000a08c7c._ewelink._tcp.local. REMOVED
2023-02-19 19:18:44.105248  | Service eWeLink_10012f59cd._ewelink._tcp.local. REMOVED
2023-02-19 19:24:11.353808  | Service eWeLink_10012f8d1b._ewelink._tcp.local. ADDED
2023-02-19 19:24:14.423450  | Service eWeLink_10012f59cd._ewelink._tcp.local. ADDED
2023-02-19 19:24:14.436138  | Service eWeLink_1000a08c7c._ewelink._tcp.local. ADDED
2023-02-19 19:27:15.899916  | Service eWeLink_1000cf7638._ewelink._tcp.local. ADDED
2023-02-19 19:27:18.768271  | Service eWeLink_1000f3e195._ewelink._tcp.local. ADDED
2023-02-19 19:46:18.220777  | Service eWeLink_1000cf7638._ewelink._tcp.local. REMOVED
2023-02-19 19:46:28.225215  | Service eWeLink_1000f3e195._ewelink._tcp.local. REMOVED
2023-02-19 19:50:28.215596  | Service eWeLink_10010a4f46._ewelink._tcp.local. REMOVED
2023-02-19 19:50:31.220321  | Service eWeLink_10012f59cd._ewelink._tcp.local. REMOVED
2023-02-19 19:55:20.780195  | Service eWeLink_10010a4f46._ewelink._tcp.local. ADDED
2023-02-19 20:00:08.377781  | Service eWeLink_100166cee5._ewelink._tcp.local. REMOVED
2023-02-19 20:00:11.377592  | Service eWeLink_10012f8d1b._ewelink._tcp.local. REMOVED
2023-02-19 20:09:24.557803  | Service eWeLink_10012f8d1b._ewelink._tcp.local. ADDED
2023-02-19 20:09:24.562682  | Service eWeLink_100166cee5._ewelink._tcp.local. ADDED
2023-02-19 20:14:18.506886  | Service eWeLink_1000a08c7c._ewelink._tcp.local. REMOVED
2023-02-19 20:17:34.862619  | Service eWeLink_1000a08c7c._ewelink._tcp.local. ADDED
2023-02-19 20:17:34.870426  | Service eWeLink_10012f59cd._ewelink._tcp.local. ADDED
2023-02-19 20:36:28.584886  | Service eWeLink_1000f21e74._ewelink._tcp.local. REMOVED
2023-02-19 20:36:31.596653  | Service eWeLink_10010a526e._ewelink._tcp.local. REMOVED
2023-02-19 20:36:34.590110  | Service eWeLink_10010a4da0._ewelink._tcp.local. REMOVED
2023-02-19 20:36:37.593229  | Service eWeLink_10010a5056._ewelink._tcp.local. REMOVED
2023-02-19 20:36:40.616009  | Service eWeLink_1000e61943._ewelink._tcp.local. REMOVED
2023-02-19 20:36:43.617862  | Service eWeLink_10012fa101._ewelink._tcp.local. REMOVED
2023-02-19 20:36:46.619726  | Service eWeLink_100166d6b6._ewelink._tcp.local. REMOVED
2023-02-19 20:36:49.628211  | Service eWeLink_10010a4f46._ewelink._tcp.local. REMOVED
2023-02-19 20:36:52.622940  | Service eWeLink_10012f8d1b._ewelink._tcp.local. REMOVED
2023-02-19 20:36:55.621758  | Service eWeLink_100166cee5._ewelink._tcp.local. REMOVED
2023-02-19 20:36:58.623011  | Service eWeLink_1000a08c7c._ewelink._tcp.local. REMOVED
2023-02-19 20:37:01.620714  | Service eWeLink_10012f59cd._ewelink._tcp.local. REMOVED
```

While ```10012f59cd``` appears as "REMOVED" it responds to INFO commands - see below.
As well executed extremely fine execute ON/OFF commands in LAN mode:
```
2023-02-19 20:41:11 [D] 10012f59cd => Local4 | 192.168.1.149:8081 | {'cmd': 'info'} <= info: 200
...
2023-02-19 20:56:12 [D] 10012f59cd => Local4 | 192.168.1.149:8081 | {'cmd': 'info'} <= info: 200
...
2023-02-19 20:57:18 [D] 10012f59cd => Local4 | 192.168.1.149:8081 | {'switch': 'off'} <= {'seq': 28, 'sequence': '1676833038000', 'error': 0}
2023-02-19 20:57:18 [D] 10012f59cd <= Local3 | 192.168.1.149:8081 | {'switch': 'off', 'startup': 'off', 'pulse': 'off', 'sledOnline': 'on', 'pulseWidth': 500, 'rssi': -83} | 28
2023-02-19 20:57:18 [D] 10012f59cd <= Cloud3 | {'switch': 'off'} | 1676833038000
2023-02-19 20:57:18 [D] 10012f59cd => Local4 | 192.168.1.149:8081 | {'switch': 'on'} <= {'seq': 29, 'sequence': '1676833038001', 'error': 0}
2023-02-19 20:57:18 [D] 10012f59cd <= Local3 | 192.168.1.149:8081 | {'switch': 'on', 'startup': 'off', 'pulse': 'off', 'sledOnline': 'on', 'pulseWidth': 500, 'rssi': -83} | 29
2023-02-19 20:57:18 [D] 10012f59cd <= Cloud3 | {'switch': 'on'} | 1676833038001
```


As soon device received, executed command and provided some feedback, it appears again:
```
2023-02-19 20:57:18.161959  | Service eWeLink_10012f59cd._ewelink._tcp.local. ADDED
```

IMHO ZC event REMOVED is not reliable source of information about device availability. 
It appears when no activity proceeded with device, still device is online and ready to use.
Unfortunately even if device responds to command INFO it gonna be REMOVED...

**Comment by KidA001** [comment]
This exact issue is happening for me, mentioned in other issues on this repo as well. Has a fix been identified I can wait for? I'm not sure I want to downgrade and risk my entities being renamed

**Comment by Masterz69** [comment]
> 

You can try PR #1116 - so far I know & using for a while: it fixing local offline's.
Then please provide feedback.

**Comment by KidA001** [comment]
@Masterz69  Glad to test. How do I point my HACS to that PR/Branch?

**Comment by Masterz69** [comment]
> @Masterz69 Glad to test. How do I point my HACS to that PR/Branch?

I know one way, but not with HACS.
Install GitHub Desktop, clone repository.
Then choose current branch - Pull requests #1116.
Now copy local repository files over HA : custom-components/sonoff over same directory.
& restart HA.

**Comment by thefunkygibbon** [comment]
hey all, a great workaround since this issue doesn't appear to be being worked on,  is to create an automation to monitor a sonoff device and restart the service automatically.  its not a perfect solution but hopefully it should make the devices work for "most" of the time. 
https://github.com/AlexxIT/SonoffLAN/issues/1138#issuecomment-1478580171

**Comment by AlexxIT** [comment]
Looks like all your ideas implemented in the v3.5.0
Thanks!

I think problem with REMOVED event started from the latest python zeroconf library. Something broken there.

But missing host is not a normal situation. I think it is your local setup problem. Maybe server OS or Router.
